File: DI/DI-Dagger-Java/src/main/java/model/named/services/ApiService.java
package model.named.services;

import model.named.IApiService;

import java.util.UUID;

abstract class ApiService implements IApiService {
    private final String id = UUID.randomUUID().toString();

    @Override
    public String toString() {
        return "ApiService{" +
                "id='" + id + '\'' +
                '}';
    }
}

File: DI/DI-Dagger-Kotlin/src/main/kotlin/models/named/services/ApiService.kt
package models.named.services

import models.named.IApiService
import java.util.*


abstract class ApiService : IApiService {
    private val id = UUID.randomUUID().toString()
    override fun toString(): String {
        return "ApiService{" +
                "id='" + id + '\'' +
                '}'
    }
}
